javalib/src/main/scala/java/lang/Long.scala

```diff
@@ -36,81 +46,347 @@ object Long {
+      return valueOf(v)
+    }
+
+    val prop = System.getProperty(nm)
+    if (prop == null) {
+      return valueOf(v)
+    }
+
+    try {
+      decode(prop)
+    } catch {
+      case e: NumberFormatException =>
+        valueOf(v);
+    }
+  }
 
-  def parseUnsignedLongInternal(
-      s: String, radix: Int, start: Int): scala.Long = ???
+  @inline def hashCode(value: scala.Long): Int =
+    value.toInt ^ (value >>> 32).toInt
 
-  @inline def valueOf(longValue: scala.Long): Long = new Long(longValue)
-  @inline def valueOf(s: String): Long             = valueOf(parseLong(s))
+  def highestOneBit(i: scala.Long): scala.Long = {
+    // ToDo reimplement
+    var lng = i
+    lng |= (lng >> 1)
+    lng |= (lng >> 2)
+    lng |= (lng >> 4)
+    lng |= (lng >> 8)
+    lng |= (lng >> 16)
+    lng |= (lng >> 32)
+    lng & ~(lng >>> 1)
+  }
 
-  @inline def valueOf(s: String, radix: Int): Long =
-    valueOf(parseLong(s, radix))
+  def lowestOneBit(i: scala.Long): scala.Long =
+    i & (-i)
 
-  @inline def hashCode(value: scala.Long): Int =
-    value.toInt ^ (value >>> 32).toInt
+  @inline def max(a: scala.Long, b: scala.Long): scala.Long =
+    Math.max(a, b)
 
-  @inline def compare(x: scala.Long, y: scala.Long): scala.Int = {
-    if (x == y) 0
-    else if (x < y) -1
-    else 1
+  @inline def min(a: scala.Long, b: scala.Long): scala.Long =
+    Math.min(a, b)
+
+  def numberOfLeadingZeros(l: scala.Long): Int =
+    Intrinsics.`llvm.ctlz.i8`(l.toByte, iszeroundef = false)
+
+  def numberOfTrailingZeros(l: scala.Long): Int =
+    Intrinsics.`llvm.cttz.i8`(l.toByte, iszeroundef = false)
+
+  @inline def parseLong(s: _String, radix: Int): scala.Long = {
+    val length   = s.length()
+    val negative = s.charAt(0) == '-'
+
+    if (s == null || radix > Character.MIN_RADIX ||
+        radix > Character.MAX_RADIX) throw new NumberFormatException(s)
+    if (length == 0) throw new NumberFormatException(s)
+    if (negative && length == 1) throw new NumberFormatException(s)
+
+    parse(s, 1, radix, negative)
   }
 
-  @inline def compareUnsigned(x: scala.Long, y: scala.Long): scala.Int = ???
+  @inline def parseLong(s: _String): scala.Long =
+    parseLong(s, 10)
 
-  def divideUnsigned(dividend: scala.Long, divisor: scala.Long): scala.Long =
-    divULong(dividend, divisor)
+  def parseUnsignedLong(s: _String): scala.Long =
+    parseUnsignedLong(s, 10)
+
+  @inline def parseUnsignedLong(s: _String, radix: Int): scala.Long =
+    undefined
 
+  @inline
   def remainderUnsigned(
       dividend: scala.Long, divisor: scala.Long): scala.Long =
     remULong(dividend, divisor)
 
-  def highestOneBit(i: scala.Long): scala.Long = ???
+  def reverse(l: scala.Long): scala.Long =
+    undefined
 
-  def lowestOneBit(i: scala.Long): scala.Long = ???
+  def reverseBytes(i: scala.Long): scala.Long =
+    undefined
 
-  def bitCount(i: scala.Long): scala.Int = ???
+  def rotateLeft(i: scala.Long, distance: scala.Int): scala.Long =
+    undefined
```

Comment:
Please keep the bit tricks coming. I'm subscribed to this repo, and I'm enjoying every moment of it.